lmfit/parameter.py

```diff
@@ -458,52 +458,34 @@ def set(self, value=None, vary=None, min=None, max=None, expr=None):
             self._val = value
         if vary is not None:
             self.vary = vary
-        if min is not None:
-            self.min = min
-        if max is not None:
-            self.max = max
+        if min is None:
+            min = -inf
+        if max is None:
+            max = inf
+        self.min = min
+        self.max = max
 
     def _init_bounds(self):
```

Comment:
`_init_bounds` is only called once, when the object is constructed. These valuable checks should be carried out whenever `min` or `max` are changed.